scripts/react-compiler-compliance-check.ts

```diff
@@ -261,15 +262,163 @@ function createFilesGlob(filesToCheck?: string[]): string | undefined {
+    // Create a map of file paths to changed line numbers for quick lookup
+    const changedLinesMap = new Map<string, Set<number>>();
+    for (const file of diffResult.files) {
+        const changedLines = new Set<number>();
+        file.addedLines.forEach((line) => changedLines.add(line));
+        file.modifiedLines.forEach((line) => changedLines.add(line));
+        changedLinesMap.set(file.filePath, changedLines);
+    }
 
-    if (success.size > 0) {
+    // Filter failures to only include those on changed lines
+    const filteredFailures = new Map<string, CompilerFailure>();
+    results.failures.forEach((failure, key) => {
+        const changedLines = changedLinesMap.get(failure.file);
+
+        // If the file is not in the diff, skip this failure
+        if (!changedLines) {
+            return;
+        }
+
+        // If the failure has a line number, check if it's in the changed lines
+        if (failure.line !== undefined) {
+            const isLineChanged = changedLines.has(failure.line);
+            if (isLineChanged) {
+                filteredFailures.set(key, failure);
+            }
+            return;
+        }
+
+        // If there's no line number, include the failure if the file has changes
+        filteredFailures.set(key, failure);
+    });
+
+    const filteredSuppressedFailures = new Map<string, CompilerFailure>();
+    results.suppressedFailures.forEach((failure, key) => {
+        if (filteredFailures.has(key)) {
+            return;
+        }
+        filteredSuppressedFailures.set(key, failure);
+    });
+
+    // Filter success set to only include files that are in the diff
+    const changedFiles = new Set(diffResult.files.map((file) => file.filePath));
+    const filteredSuccesses = new Set<string>();
+    results.success.forEach((file) => {
+        if (!changedFiles.has(file)) {
+            return;
+        }
+        filteredSuccesses.add(file);
+    });
+
+    if (shouldPrintSuccesses) {
+        if (filteredSuccesses.size === 0) {
+            logInfo('No successes remain after filtering by diff.');
+        } else {
+            logInfo(`${filteredSuccesses.size} out of ${results.success.size} successes remain after filtering by diff.`);
+        }
+    }
+
+    if (shouldPrintSuppressedErrors) {
+        if (filteredSuppressedFailures.size === 0) {
+            logInfo('No suppressed errors remain after filtering by diff.');
+        } else {
+            logInfo(`${filteredSuppressedFailures.size} out of ${results.suppressedFailures.size} successes remain after filtering by diff.`);
+        }
+    }
+
+    if (filteredFailures.size === 0) {
+        logInfo('No failures remain after filtering by diff.');
+    } else {
+        logInfo(`${filteredFailures.size} out of ${results.failures.size} failures remain after filtering by diff.`);
+    }
+
+    return {
+        success: filteredSuccesses,
+        failures: filteredFailures,
+        suppressedFailures: results.suppressedFailures,
```

Comment:
I think you want `filteredSupressedFailures` here?


```suggestion
        suppressedFailures: filteredSuppressedFailures,
```